### ISSUE TITLE
default network

### ISSUE THREAD

**Issue by vieux** [docker/compose#2292]
I'm not sure I'm reading correctly, but it seems like compose is choosing `bridge` by default as default network driver : https://github.com/docker/compose/blob/master/compose/project.py#L86

We should let the daemon decide instead, same as this PR https://github.com/docker/docker/pull/17504

**Comment by vieux** [comment]
ping @dnephin

**Comment by dnephin** [comment]
Yup, I agree

**Comment by dnephin** [comment]
I've got a fix in #2295. It works fine against engine. When I run it against swarm, it creates an overlay network, but then I get this failure trying to start the container:

```
docker.errors.APIError: 500 Server Error: Internal Server Error (
  "b'Cannot start container 3f5f713dd01ae90ba420f77f47abb040360d2360c851b027c049ccc89736e53e:
  subnet sandbox join failed for "10.0.0.0/24":
  vxlan interface creation failed for subnet "10.0.0.0/24":
  failed in prefunc:
  failed to set namespace on link "vxlan9323b4a": invalid argument'")
```
